### ISSUE TITLE
Poor ML Algorithm selection

### ISSUE THREAD

**Issue by KowshikB03** [AOSSIE-Org/Social-Street-Smart#80]
For Hate Speech detection, the dataset has only 1 meaningful feature "comment_text" (excluding the ID) to be trained on. CNN and LSTM are the algorithms developed for this dataset. However,

1. the problem is not complex enough to require the power of CNNs or LSTMs.
2. Neural networks excel only when there are many features or complex patterns in the data (e.g., image, text, or time-series data).
3. Neural networks and LSTMs are computationally expensive to train on simpler datasets with less than 5 features.
4. Simpler models like Random Forest is easier to interpret and debug compared to neural networks.

Hence I developed a Random Forest model with benefits like,

1. Computational Efficiency i.e less training time (6 minutes [RF model] < 15+ minutes [CNNs & LSTM])
2. Mean AUC of RF model (0.9655) is higher than both CNN(0.9609) and LSTM(0.9588)
3. Achieve same validation accuracy (0.994) as both CNN and LSTM with less training time

Hence I request to merge better performing RF model in addition to CNN and LSTM.
